django/db/models/base.py

```diff
@@ -451,18 +453,34 @@ class ModelState(object):
     """
     A class for storing instance state
     """
+
+    __slots__ = ['db', 'adding']
+
```

Comment:
I've personally used and seen third party apps use `_state` to store extra details about model instances.

For example, it can be useful to store the schema the instance was retrieved from in PostgreSQL multi-tenancy app.